### PR TITLE
feat: create file from template

### PR DESCRIPTION
**Pull request by kulmann**
## Description
Adding a new file action for creating documents from templates via app provider (hence placing the action in the `external` app).

## Related Issue
- Fixes https://github.com/owncloud/web/issues/11750

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:
- [x] re-prioritize file actions: `create from template` needs to be first action (on left click and in right-click context menu)
- [x] Closing the created document needs to navigate to the personal space root, not to the location of the template file
- [x] Changelog
- [x] Error handling during handler execution
- [ ] Test coverage

### CODE REVIEW

**Comment by JammingBen** [comment]
One bug that we found: when opening a template file in a project space while having the sidebar open, the shares panel still shows the project space members, despite the new file being in the personal space. This happens because the ancestors and therefore the loaded space members don't get cleared. This is a general problem with the app wrapper.

File: packages/web-app-external/src/extensions/createFromTemplate.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+      const template = resources[0]
+      if (!template.canDownload()) {
+        return false
+      }
+
+      const templateMimeType = appProviderService.templateMimeTypes.find(
+        (mimeType) => mimeType.mime_type === template.mimeType
+      )
+      return !!templateMimeType
```
Comment (JammingBen) [inline packages/web-app-external/src/extensions/createFromTemplate.ts:57]:
```suggestion
      return appProviderService.templateMimeTypes.some(
        (mimeType) => mimeType.mime_type === template.mimeType
      )
```

File: packages/web-app-external/src/extensions/createFromTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          [contextRouteParamsKey]: { driveAliasAndItem: spacesStore.personalSpace.driveAlias },
+          [contextRouteQueryKey]: { fileId: spacesStore.personalSpace.fileId }
+        }
+        routeOptions.query = {
+          ...routeOptions.query,
+          ...contextRouteOptions
+        }
+        await router.push(routeOptions)
+      } catch (e) {
+        // TODO: can we get more specific with error titles?
```
Comment (JammingBen) [inline packages/web-app-external/src/extensions/createFromTemplate.ts:114]: IMO it's fine to have it generic for the user here.

**Review by JammingBen — COMMENTED** [review]
Few nitpicks, rest is awesome!

**Review by JammingBen — APPROVED** [review]

**Comment by saw-jan** [comment]
Test added in https://github.com/owncloud/web/issues/11811

**Pull request merged**